SIV_scraper

```diff
@@ -0,0 +1,215 @@
+                                    hr(),
+                                    actionButton("submitSIV", "Submit"),
+                                    downloadButton("dl_SIV", "Download"),
+                                    hr(),
+                                    DTOutput("SIV_view"))
+                                             
+                )#end of navbar page
+)#end of UI
+
+
+server <- function(input, output, session) {
+  
+  address=function(){
+    page <- read_html("Data/temp_files/scrapedpage.html")
+    page %>%
+      html_nodes("a") %>%       # find all links
+      html_attr("href") %>%     # get the url
+      str_subset("\\.pdf") %>% # find those that end in xlsx
+      .[[5]]                    # look at the first one
+
+  }
+
+  SIV_data=reactiveFileReader(
+    intervalMillis=86400000,
+    session=session,
+    filePath=address(),
+    readFunc = function(filePath) {
+       download.file(filePath, destfile="Data/temp_files/SIV.pdf", mode="wb")
+      data = pdf_data("Data/temp_files/SIV.pdf")
+      data <- data[[1]]
+      data
+    }
+  )
+  # SIV_data = reactive({
+  #   data = pdf_data("Data/temp_files/SIV.pdf")
+  #   data <- data[[1]]
+  #   data
+  # 
+  # })
+  ##Reactive to find data date
+  SIV_date = reactive({
+  SIV_date_row <- SIV_data()[grep("Date:", SIV_data()$text),4]
+  SIV_data_date <- SIV_data() %>% filter(y == as.numeric(SIV_date_row))
+  SIV_data_date$text <- as.Date(SIV_data_date$text, format = "%d/%m/%Y") 
+  SIV_data_date <- na.omit(SIV_data_date)
+  SIV_data_date <- SIV_data_date %>% dplyr::pull(text) 
+  SIV_data_date})
+  
+
+  SIV_data_clean = reactive({
+  ##Remove comment
+  #Remove starting rows
+  meta_row <- SIV_data()[grep("Product", SIV_data()$text), 4]
+  daily_data <- SIV_data() %>% filter(y >= as.numeric(meta_row))
+  
+  ##Clean data
+  find_cols <- daily_data %>% filter(text == "Product"| text == "Designation"| text == "Origin"| text == "Country"| text == "Market"| text == "Value"| text == "Quantity")  
+  find_cols <- find_cols$x - c(1,13,8, 36, 21, 1, 1) 
+  find_rows <- unique(daily_data$y)
+  find_rows <- find_rows[c(-1)]
+  find_start <- min(daily_data$x)-1
+  
+  ##Turn into table
+  clean_data <-  daily_data %>% mutate(col = cut(x, breaks = c(0, find_cols, Inf))) %>% 
+    mutate(row = cut(y, breaks = c(0, find_rows, Inf))) %>%
+    filter(col != "(0,39]") %>%
+    filter(col != "(39,92]") %>%
+    arrange(col, row) %>% 
+    group_by(col, row) %>% 
+    dplyr::mutate(text = paste(text, collapse = " ")) %>% 
+    ungroup() %>%
+    select(row, text, col) %>% 
+    unique() %>% 
+    spread(col, text) %>%
+    select(-row)
+  
+  ##Change col names
+  colnames(clean_data) <- clean_data[1,]
+  clean_data <- clean_data[-1,]
+  clean_data <- clean_data[which(!is.na(colnames(clean_data)))]
+  clean_data <- clean_data %>% select(Designation, `Origin Code`, Country, `Market Place`, Value, Quantity)
```

Comment:
Check your line spacing; code is easier to read if lines aren't too long. There is a guide in rstudio which tells you when lines are too long!